src/sentry/tasks/assemble.py

```diff
@@ -311,14 +326,91 @@ def _create_artifact_bundle(
                     artifact_bundle=artifact_bundle,
                     source_file_type=source_file_type.value,
                 )
+        else:
+            raise AssembleArtifactsError(
+                "uploading a bundle without debug ids or release is prohibited"
+            )
+
+
+def get_with_manifest_fallback(value, manifest_key, manifest, throw=True):
+    if value is None:
+        manifest_value = manifest.get(manifest_key)
+        if manifest_value is None and throw:
+            raise AssembleArtifactsError(f"failed to load {manifest_key} from the manifest")
+
+        return manifest_value
+
+    return value
+
+
+def handle_assemble_for_release_file(bundle, archive, manifest, organization, version):
+    org_slug = manifest.get("org")
+    if organization.slug != org_slug:
+        raise AssembleArtifactsError("organization does not match uploaded bundle")
+
+    release_name = manifest.get("release")
+    if release_name != version:
```

Comment:
I found this strange too, I just kept the old code in order to avoid regressions (maybe during renaming some problems would come up). All of this under the assumption that we are going to fade it out.

I can follow up with a PR to actually improve all the code but I am not sure it's worth it.